command-line/lib/Command/Key/Public.hs

```diff
@@ -40,13 +43,19 @@ mod liftCmd = command "public" $
         <> progDesc "Get the public counterpart of a private key"
         <> footerDoc (Just $ string $ mconcat
             [ "The private key is read from stdin."
+            , "Extended public key is default. In order to get public key"
+            , ", ei., without chain code use '--pub'."
```

Comment:
Do you know which use case is more common?
I am thinking that public key only might be a better default. Not sure though.
Another alternative could be to have completely separate subcommands -- e.g. `public` and `xpub`, or something like that. Then you could tell users to run `xpub` if they need to derive child keys.

Reply:
More than which one is most common, I'd go for the "safest" from a privacy standpoint here. Since sharing your chain code allows other to create new (public) child keys from your public key, you may inadvertently gives access to an entire part of your sub-tree by sharing your extended public key. 

Plus, cardano-cli and other cardano-tools mostly deals with non-extended ed25519 keys only.

Reply:
introduced `--with-chain-code` and `--without-chain-code` here https://github.com/input-output-hk/cardano-addresses/pull/87/commits/c2e3c7801beb75a5541c86a1a3d8e09f06d17459